src/libsystemd/sd-id128/id128-util.h

```diff
@@ -24,6 +24,11 @@ int id128_read(const char *root, const char *p, Id128FormatFlag f, sd_id128_t *r
 int id128_write_fd(int fd, Id128FormatFlag f, sd_id128_t id);
 int id128_write(const char *p, Id128FormatFlag f, sd_id128_t id);
 
+int id128_get_machine_impl(const char *root, sd_id128_t *ret);
+static inline int id128_get_machine(const char *root, sd_id128_t *ret) {
+        return root ? id128_get_machine_impl(root, ret) : sd_id128_get_machine(ret);
+}
```

Comment:
Why have two functions? Can't we just have one that calls `sd_id128_get_machine()` if `root == NULL`?

Reply:
`sd_id128_get_machine()` caches the result. When `root` is specified, the result should not be cached.